### DISCUSSION TITLE
Which API shows that the collective operations include barrier, all-to-one, all-to-all, and reduction operations?

### DISCUSSION THREAD

**Discussion by XCloudAI** [openucx/ucx#6949] [Q&A] [answered]
Below page says that ```The collective operations include barrier, all-to-one, all-to-all, and reduction operations. When possible, we will take advantage of hardware acceleration for collectives (e.g., InfiniBand Switch collective acceleration).```
https://openucx.github.io/ucx/api/v1.10/html/md__hpc_mtr_scrap_users_yosefe_ucx_docs_doxygen_design.html#UCP
Which API shows this capability? Any example code to show how to use them?

**Comment by shamisp** [comment, marked as answer]
This capability was spawn into the separate project: https://github.com/openucx/ucc
